Mix of various upstream commits to build on FreeBSD and be more POSIX compliant.
Drop at next bump.


Index: src/lib/url.c
===================================================================
--- src/lib/url.c	(revision 308)
+++ src/lib/url.c	(revision 309)
@@ -26,6 +26,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
+#include <netinet/in.h>
 
 #include "misc.h"
 #include "url.h"
Index: src/lib/proxy_factory.c
===================================================================
--- src/lib/proxy_factory.c	(revision 308)
+++ src/lib/proxy_factory.c	(revision 309)
@@ -27,6 +27,7 @@
 #include <math.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
+#include <netinet/in.h>
 #include <pthread.h>
 
 
Index: src/plugins/pacrunner_webkit.c
===================================================================
--- src/plugins/webkit.c	(revision 335)
+++ src/plugins/webkit.c	(working copy)
@@ -23,6 +23,7 @@
 #include <sys/socket.h>
 #include <netdb.h>
 #include <arpa/inet.h>
+#include <netinet/in.h>
 #define __USE_BSD
 #include <unistd.h>
 
Index: src/plugins/pacrunner_mozjs.c
===================================================================
--- src/plugins/mozjs.c	(revision 335)
+++ src/plugins/mozjs.c	(working copy)
@@ -23,6 +23,7 @@
 #include <sys/socket.h>
 #include <netdb.h>
 #include <arpa/inet.h>
+#include <netinet/in.h>
 #define __USE_BSD
 #include <unistd.h>
 
Index: configure.ac
===================================================================
--- configure.ac	(revision 334)
+++ configure.ac	(revision 335)
@@ -199,7 +199,7 @@
 AC_TYPE_SIZE_T
 PLUGINDIR=$libdir/$PACKAGE_NAME/$PACKAGE_VERSION/plugins
 AC_SUBST(PLUGINDIR)
-CFLAGS="-g -std=c99 $CFLAGS -DPLUGINDIR=\\\"$PLUGINDIR\\\" -DSYSCONFDIR=\\\"$sysconfdir\\\" -D_POSIX_C_SOURCE=1"
+CFLAGS="-g -std=c99 $CFLAGS -DPLUGINDIR=\\\"$PLUGINDIR\\\" -DSYSCONFDIR=\\\"$sysconfdir\\\" -D_POSIX_C_SOURCE=200112L"
 
 ### Checks for library functions.
 AC_FUNC_MALLOC
Index: src/bin/Makefile.am
===================================================================
--- src/bin/Makefile.am	(revision 318)
+++ src/bin/Makefile.am	(revision 319)
@@ -3,5 +3,4 @@
 # Command line interface to libproxy
 proxy_SOURCES = proxy.c
 proxy_CFLAGS  = -I$(top_srcdir)/src/lib
-proxy_LDFLAGS = -ldl
 proxy_LDADD   = ../lib/libproxy.la
